### PR TITLE
task-56773 add structure for timeseries-generator

### PR DESCRIPTION
**Pull request by alexdinescu**

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# but only works if all arguments require a hyphenated flag
+# -v; -SL; -f arg; etc will work, but not arg1 arg2
+if [ "$#" -eq 0 ] || [ "${1#-}" != "$1" ]; then
+    set -- java -jar /app.jar "$@"
+fi
+
+# check for the expected command
+if [ "$1" = 'java']; then
+    # use gosu (or su-exec) to drop to a non-root user
+    exec gosu java -jar /app.jar "$@"
```
Comment (alexdinescu) [inline entrypoint.sh]: not sure this works, to be checked

File: src/main/java/com/tradingcentral/timeseries/generator/Application.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.tradingcentral.timeseries.generator;
```
Comment (alexdinescu) [inline src/main/java/com/tradingcentral/timeseries/generator/Application.java]: this package name looks ugly ( too long words ).
does anyone has a better naming alternative ?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+FROM openjdk:11-slim
+COPY --from=builder /home/gradle/src/build/libs/timeseries-generator.jar app.jar
+COPY --from=builder /home/gradle/src/entrypoint.sh .
+RUN chmod +x entrypoint.sh
+
+COPY --from=builder /home/gradle/src/build/test-results/test test-results/
+
+# Run main
+EXPOSE 5000
+ENTRYPOINT ["/entrypoint.sh"]
```
Comment (migescu) [inline Dockerfile]: we should add ` -XX:MaxRAMPercentage=85.0` here too

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+dependencies {
+	annotationProcessor 'org.projectlombok:lombok:1.18.4'
+
+	compileOnly 'org.projectlombok:lombok:1.18.4'
+
+	compile group: 'org.springframework.cloud', name: 'spring-cloud-context', version: '2.1.3.RELEASE'
+	compile group: 'org.springframework.cloud', name: 'spring-cloud-starter-aws-parameter-store-config', version: '2.1.3.RELEASE'
+
+	compile group: 'org.yaml', name: 'snakeyaml', version: '1.24'
```
Comment (migescu) [inline build.gradle]: do we need this?

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	compile group: 'org.apache.commons', name: 'commons-text', version: '1.6'
+	compile group: 'io.swagger', name: 'swagger-jersey2-jaxrs', version: '1.5.13'
+
+	compile('org.springframework.boot:spring-boot-starter-jersey')
+	compile('org.springframework.boot:spring-boot-starter')
+	compile('org.springframework.boot:spring-boot-starter-web')
+
+	implementation('org.springframework.boot:spring-boot-starter-data-jpa')
+	runtimeOnly('org.postgresql:postgresql:42.2.6')
```
Comment (migescu) [inline build.gradle]: not sure that we need this either

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# service generator API port
+server.port = 5000
+
+# non-web app
+spring.main.web-application-type=NONE
+
+spring.jpa.database-platform=org.hibernate.dialect.PostgreSQLDialect
+spring.jpa.properties.hibernate.temp.use_jdbc_metadata_defaults=false
+spring.jpa.show-sql=${SHOW_SQL:false}
```
Comment (migescu) [inline src/main/resources/application.properties]: I don't think we need these either

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# service generator API port
+server.port = 5000
+
+# non-web app
+spring.main.web-application-type=NONE
+
```
Comment (migescu) [inline src/main/resources/application.properties]: we should add the logging level

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+spring.jpa.database-platform=org.hibernate.dialect.PostgreSQLDialect
+spring.jpa.properties.hibernate.temp.use_jdbc_metadata_defaults=false
+spring.jpa.show-sql=${SHOW_SQL:false}
+
+# AWS Param Store overrides - enable for local testing
```
Comment (migescu) [inline src/main/resources/application.properties]: cleanup

File: src/test/java/com/tradingcentral/timeseries/generator/ApplicationContextLoadTest.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+@Slf4j
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+    classes = Application.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = {"aws.paramstore.enabled=false"})
+public class ApplicationContextLoadTest {
+  @Test
+  public void applicationContextLoadTest() {
+    log.info("great test");
```
Comment (migescu) [inline src/test/java/com/tradingcentral/timeseries/generator/ApplicationContextLoadTest.java]: 👍

**Pull request merged**